uiowa/uiowa
# ITS Sprint 232 Updates

Resolves https://github.com/uiowa/uiowa/issues/8087
Resolves https://github.com/uiowa/uiowa/issues/5609





# ToDo

- [x] Confirm permissions are correct for ITS editor, publisher, help desk roles
- [x] Make sure workflows/workbench email all still work as expected based on roles changing
- [x] Check/update permissions on Admissions and Grad related to fragments
- [x] Check docs about fragments mentioning permissions
- [x] Write up summary of permission adjustments for ITS to review
- [x] Deploy branch to DEV for ITS to review

# Summary of ITS website permissions

Here is an overview of the default permissions:
https://sitenow.uiowa.edu/documentation/user-management-and-roles

_We will be making a change to the default permissions that editors can no longer work with fragments (pre-footer, after content)._

## Editor

Includes SiteNow defaults except:

- Create/edit/delete any support article with the draft/publish/archive workflow transitions. Viewing/reverting/deleting revisions is also included.
- Create/edit/replicate any alert with the draft/submit workflow transitions. Viewing/reverting revisions is also included.
- Edit any service with the draft/publish workflow transitions. Viewing/reverting/deleting revisions is also included.
- Cannot manage main menu items
- Cannot create/edit/delete/revert pages
- Cannot edit the footer contact info block

## Publisher (still expected to be stacked on top of Editor role)

Includes SiteNow defaults except:

- Override "authored by" on support articles
- Cannot work with fragments (pre-footer, after content)
- Cannot edit the footer contact info block

## Help Desk (still expected to be stacked on top of Editor role)

Similar to publisher on SiteNow default except:

- Create/edit/delete/replicate any alert with the draft/submit/publish/archive workflow transitions. Viewing/reverting/deleting revisions is also included.
- Allows sending an email on alert save.
- Cannot create/edit/delete/revert pages
- Cannot manage main menu items
- Cannot work with fragments (pre-footer, after content)
- Cannot edit the footer contact info block

# How to test

```
ddev blt ds --site=its.uiowa.edu
```

- Go to /admin/people to see the list of users. There should be no editor/publisher stacked users. Instead there should be editor/help-desk stacked users.
- Feel free to check the permissions as highlighted above, but this has already been confirmed by ITS.

```
ddev blt ds --site=admissions.uiowa.edu
```

- Confirm that the editor role can still reach the fragments overview page and can create the site-specific fragment, "program"
